src/structs/lepton_encoder.rs

```diff
@@ -658,10 +672,10 @@ fn roundtrip_large_coef() {
     // to the binary format of the compressor and probably shouldn't do that since then
     // the compressor and decompressor won't be compatible anymore.
     let verified_output = [
-        0x506f55523369cf48,
-        0x5b2795bc24a04d2e,
-        0xdcb68ed904cfc4f9,
-        0x80efab28c62db071,
+        0x3195FA3A5611FE57,
```

Comment:
why have these changed?

Reply:
To reduce number of operations in the new scheme eob now is (7 - old eob), then branches in corresponding part of the model are mirrored. Therefore hash is different.

Reply:
Since using the mask it doesn't seem up anything else significantly, maybe we can go back to the old code for the eob calcs... this is already a pretty big change and if we can minimize the potential regressions and just have the changes strictly necessary for this change.

Reply:
OK, I'll prepare such a variant and check performance then.

Reply:
Merged the changes and left out the mask changes for now. We can do this as a separate change. 

Since this is a pretty major change, I was wondering if you could add some more comments explaining how this work.

Reply:
Thanks for your effort to remove nonzero_mask. Added some comments (and removed no longer needed ones).